src/dss/utils.py

```diff
@@ -50,6 +62,21 @@ def wait_for_deployment_ready(
             logger.info(f"Deployment {deployment_name} in namespace {namespace} is ready")
             break
         elif time.time() - start_time >= timeout_seconds:
+            # Surround the following block with try-except?
+            # ----Block-start----
+            pod: Pod = list(
+                client.list(
+                    Pod,
+                    namespace=namespace,
+                    labels={"canonical.com/dss-notebook": deployment_name},
+                )
+            )[0]
+            reason = pod.status.containerStatuses[0].state.waiting.reason
+            if reason in ["ImagePullBackOff", "ErrImagePull"]:
+                raise ImagePullBackOffError(
+                    f"Failed to create Deployment {deployment_name} with {reason}"
```

Comment:
I think in this case we should still mention timeout.  It isn't that this failed because we're in ImagePullBackOff, its that we timed out and at the end we see we're still in that status.  Sometimes ImagePullBackOff is a transient thing, so I suggest we're clear that isn't what we think happened

```suggestion
                    f"Timed out trying to create Deployment {deployment_name} with {reason}"
```

Reply:
I agree with the timeout but based on the UX team feedback we should avoid using tterms like Pod and Deployment when user is actually creating notebooks. We can have our own debug message in logger.debug with the pods and deployments but for the user error message we should rather avoid that. Moreover another UX recommendation is always to provide recommended next steps on errors if there are some (writing it here maybe it does not apply).

Reply:
I changed it to say timeout and logged this in the debug log, the error log prints `Notebook` not `Deployment`